apps/explorer/src/components/ownedobjects/views/DynamicFieldsCard.tsx

```diff
@@ -0,0 +1,131 @@
+        [fetchNextPage, hasNextPage, isFetchingNextPage]
+    );
+
+    useEffect(() => {
+        const element = observerElem.current;
+        if (!element) return;
+        const option = { threshold: 0 };
+        const observer = new IntersectionObserver(handleObserver, option);
+        observer.observe(element);
+        return () => observer.unobserve(element);
+    }, [fetchNextPage, hasNextPage, handleObserver]);
+
+    if (isInitialLoading) {
+        return (
+            <div className="mt-1 flex w-full justify-center">
+                <LoadingSpinner />
+            </div>
+        );
+    }
+
+    // show the dynamic fields tab if there are pages and the first page has data
+    const hasPages = !!data?.pages?.[0].data.length;
+
+    return hasPages ? (
+        <div className="mt-10">
+            <TabGroup size="lg">
+                <TabList>
+                    <Tab>Dynamic Fields</Tab>
+                </TabList>
+                <TabPanels>
+                    <TabPanel>
+                        <div className="mt-4 flex max-h-[600px] flex-col gap-5 overflow-auto overflow-x-scroll">
+                            {data.pages.map(({ data }) =>
+                                data.map((result) => (
+                                    <DisclosureBox
+                                        title={
+                                            <div className="min-w-fit max-w-[60%] truncate break-words text-body font-medium leading-relaxed text-steel-dark">
+                                                {result.name?.value.toString()}:
+                                            </div>
+                                        }
+                                        preview={
+                                            <div className="flex items-center gap-1 break-all">
+                                                <ObjectLink
+                                                    objectId={result.objectId}
+                                                />
+                                            </div>
+                                        }
+                                        variant="outline"
+                                        key={result.objectId}
+                                    >
+                                        <div className="flex flex-col divide-y divide-gray-45">
+                                            <SyntaxHighlighter
+                                                code={JSON.stringify(
+                                                    result,
+                                                    null,
+                                                    2
+                                                )}
+                                                language="json"
+                                            />
+                                            <UnderlyingObjectCard
+                                                parentId={result.objectId}
```

Comment:
we should use `id` here. It should be the parent id not the id of the object

Reply:
this should fix the problem with getDynamicFieldObject here https://github.com/MystenLabs/sui/pull/10386/files#r1163321069